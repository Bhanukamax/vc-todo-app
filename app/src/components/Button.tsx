import React from "react";
import styled from "styled-components";

interface IProps {
  active?: boolean;
  children: any;
  onClick?: any;
}

export const Button: React.FC<IProps> = ({
  active,
  children,
  ...rest
}: IProps) => (
  <StyledButton active={active} {...rest}>
    {children}
  </StyledButton>
);

const StyledButton = styled.button.attrs((): any => ({}))`
  border: 0;
  padding: 10px 20px;
  margin: 0 10px;
  border-radius: 25px;
  font-size: 1em;
  color: ${props => (props.active ? "#fff" : "#666")};
  background: ${props => (props.active ? "#83daca" : "#eaeaea")};
  :hover {
    background: #83daca;
    color: #fff;
    cursor: pointer;
  }
`;
