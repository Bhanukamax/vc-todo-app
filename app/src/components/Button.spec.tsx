import React from "react";
import { Button } from "./Button";
import { shallow } from "enzyme";

describe("<Button/>", () => {
  it("renders correctly wihtout children", () => {
    const component = shallow(<Button />);
    expect(component).toMatchSnapshot();
  });

  it("renders the button correctly with children", () => {
    const component = shallow(
      <Button>
        <p>test</p>
      </Button>
    );
    expect(component).toMatchSnapshot();
  });
});
