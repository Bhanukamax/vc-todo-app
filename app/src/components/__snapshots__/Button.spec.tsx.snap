// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> renders correctly wihtout children 1`] = `<styled.button />`;

exports[`<Button/> renders the button correctly with children 1`] = `
<styled.button>
  <p>
    test
  </p>
</styled.button>
`;
